matching/main.js

```diff
@@ -1,475 +0,0 @@
-                }
-
-                // put the mapping now
-                if (existence) {
-                    client.indices.putMapping({
-                        index: options.indexName,
-                        type: options.docType,
-                        body: mapping
-                    }, function (err, resp, status) {
-                        if (err) {
-                            console.log('mapping error: ' + err.message);
-                            return callback(err)
-                        } 
-                        console.log("mapping loaded");
-                        return callback(null, true);
-                    });
-                }
-            }
-        ], (err, results) => {
-            if (err) {
-                console.log('setting error: ' + err);
-            }
-
-            if (options.action === "index") {
-                // launch the heavy indexing stuff...
-                index(options);
-            }
-        })
-    }
-
-}
-
-/**
- * This function removes some non-used stuff from a crossref work entry,
- * in particular the citation information, which represent a considerable
- * amount of data.
- */
-function massage(data) {
-    var jsonObj = JSON.parse(data);
-    return jsonObj;
-}
-
-function buildBibliographicField(obj) {
-    var res = "";
-
-    if (obj.author)
-        res += obj.author;
-    else if (obj.first_author)
-        res += obj.first_author;
-
-    if (obj.title)
-        res += " " + obj.title;
-
-    if (obj.journal)
-        res += " " + obj.journal;
-
-    if (obj.abbreviated_journal)
-        res += " " + obj.abbreviated_journal;
-
-    if (obj.volume)
-        res += " " + obj.volume;
-
-    if (obj.issue)
-        res += " " + obj.issue;
-
-    if (obj.first_page)
-        res += " " + obj.first_page;
-
-    if (obj.year)
-        res += " " + obj.year;
-
-    return res.trim();
-}
-
-function filterType(doc) {
-    // if document type is component, we ignore it (it means that the DOI is 
-    // for a sub-part of a publication, like 10.1371/journal.pone.0104614.t002)
-    if (doc.type === "component") {
-        return true;
-    }
-    return false;
-}
-
-function index(options) {
-    var readStream = fs.createReadStream(options.dump)
-        .pipe(lzma.createDecompressor())
-        .pipe(es.split())
-        .pipe(es.map(function (data, cb) {
-            // prepare/massage the data
-            //console.log(data);
-            data = massage(data);
-            var obj = new Object();
-
-            // - migrate id from '_id' to 'id'
-            obj._id = data._id.$oid;
-            delete data._id;
-
-            // Just keep the fields we want to index
-
-            // - Main fields (in the mapping)
-            obj.title = data.title;
-            obj.DOI = data.DOI;
-
-            if (data.author) {
-                obj.author = "";
-                for (var aut in data.author) {
-                    if (data.author[aut].sequence === "first") {
```

Comment:
I asked crossref support about this, because this 'sequence' attribute is not available in the last dump, maybe consider the first one in the liste as first too, also the language attribut is not available (maybe it is not relevant for you..)